### PR TITLE
fix persistence bugs

### PR DESCRIPTION
**Pull request by alorenzo175**
- [ ] Closes #xxxx .
  - [ ] I am familiar with the [contributing guidelines](https://solarforecastarbiter-core.readthedocs.io/en/latest/contributing.html).
  - [ ] Tests added.
  - [ ] Updates entries to [`docs/source/api.rst`](https://github.com/SolarArbiter/solarforecastarbiter-core/blob/master/docs/source/api.rst) for API changes.
  - [ ] Adds descriptions to appropriate "what's new" file in [`docs/source/whatsnew`](https://github.com/SolarArbiter/solarforecastarbiter-core/tree/master/docs/source/whatsnew) for all changes. Includes link to the GitHub Issue with `` :issue:`num` `` or this Pull Request with `` :pull:`num` ``. Includes contributor name and/or GitHub username (link with `` :ghuser:`user` ``).
  - [ ] New code is fully documented. Includes [numpydoc](https://numpydoc.readthedocs.io/en/latest/format.html) compliant docstrings, examples, and comments where necessary.
  - [ ] Maintainer: Appropriate GitHub Labels and Milestone are assigned to the Pull Request and linked Issue.

### CODE REVIEW

**Comment by alorenzo175** [comment]
One problem with using index persistence by default for the persistence forecasts is the mismatch that is likely between the calculated profile and the "actual" clear profile. For the RTC ABQ site, the forecasts will basically look like. It is a fine benchmark that users should be able to improve on, but it's ugly. What do  you think @wholmgren 

image

**Comment by wholmgren** [comment]
I think the morning is bad because the sun rises behind the array, stays there for a couple of hours at this time of year, and we're not modeling that well. Still, I'm confused about why it looks that bad because the `scalar_index` function limits the index to the range [0, 2]. What does the 5 minute ahead forecast look like?

**Comment by alorenzo175** [comment]
image

**Comment by wholmgren** [comment]
Well that definitely doesn't look like it's limiting the clear sky index in the way I expect. The graph/metadata also makes me realize I didn't look carefully enough at the template persistence parameters - I thought the 5 minute and 15 minute interval length forecasts would also have 5/15 run lengths.

Maybe we should merge this PR and address the potential index issue separately.

**Comment by alorenzo175** [comment]
The 5/15 do have the appropriate run_lengths. I was just messing around manually and forgot to change that. 

image

**Comment by alorenzo175** [comment]
image

**Comment by alorenzo175** [comment]
At this time, I don't think there is an issue in the persistence code. We can merge this, but I want to make sure we agree that index persistence vs measurement persistence should be used for the reference forecasts. It will be a pain to switch once we start making them

**Comment by wholmgren** [comment]
Thanks. The 5 minute looks reasonable. I still don't understand how the hourly forecast is more than twice the clear sky expectation.

**Comment by alorenzo175** [comment]
Ah I think there are a few issues going on. https://gist.github.com/alorenzo175/b94b129c41a2afde79abb1b5fc3753d3#file-index_persistence_check-ipynb

One is that I think the timezone of the site might be wrong. Seems like it might be on MST year round. Another issue might be the modelling parameters, so the spike isn't actually over 2 the clear value.

**Comment by alorenzo175** [comment]
I think we'll leave things like they are then in terms of how the reference persistence forecasts are defined.

File: solarforecastarbiter/io/reference_observations/tests/test_common.py [hunk 1]
```diff
@@ -772,10 +772,13 @@ def test_create_forecasts(template_fx, mocker):
     api.list_observations = mocker.MagicMock(
         return_value=[site_test_observation.replace(site=site)])
     create_nwp = mocker.spy(common, 'create_nwp_forecasts')
+    create_perst = mocker.spy(common, 'create_persistence_forecasts')
     fxs = common.create_forecasts(api, site, vars_, templates)
     assert len(fxs) == 3
     assert create_nwp.call_count == 1
+    assert create_perst.call_count == 1
     assert create_nwp.call_args[0][-1] == templates[:1]
+    assert create_perst.call_args[0][-1] == templates[-1:]
```
Comment (wholmgren) [inline solarforecastarbiter/io/reference_observations/tests/test_common.py:781]: why is this line `[-1:]` and the line above `[:1]`? Seems like `[1]`, and `[0]` would work, but maybe I'm missing something.

Comment (alorenzo175) [reply solarforecastarbiter/io/reference_observations/tests/test_common.py:781]: they have to be lists. could  have done ``create_perst.call_args[0][-1][0] == templates[-1]`` instead

**Pull request merged**